### PR TITLE
EmailManager

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): #1965 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

`EmailManager` is not supported.

## What is the new behavior?

`EmailManager` is supported with platform-specific implementation on Android, iOS and `mailto:`  on other platforms.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [x] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+namespace Windows.ApplicationModel.Email
+{
+	public partial class EmailManager
+	{
+		private const int HandlerPickerResultCode = 0;
+
+		private static async Task ShowComposeNewEmailInternalAsync(EmailMessage message)
+		{
+			if (message is null)
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs]:
```suggestion
			message = message ?? throw new ArgumentNullException(nameof(message));
```

File: src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		private static async Task<ComponentName> ResolveEmailComponent()
+		{
+			var packageManager = Application.Context.PackageManager;
+
+			var emailDummyIntent = new Intent(Intent.ActionSendto);
+			emailDummyIntent.SetData(Android.Net.Uri.Parse("mailto:dummy@email.com"));
+
+			var emailActivities = packageManager.QueryIntentActivities(emailDummyIntent, 0);
+
+			if (emailActivities == null) return null;
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs]: Braces

File: src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs]: 👌

**Comment by MartinZikmund** [comment]
What's missing is currently the support for attachments. From my quick research Andorid and iOS support file-based attachments only, however UWP has this API fully based on streams. There would be two options:

- Take the stream, write to file, use that file as an attachment, then clean it up "sometime" afterward - iI don't really like this though, as it means the API would eat up storage space, which could potentially be in MB range, and there would have to be a check to clean it up even in case something fails.
- Add some Uno-specific methods, that would allow adding file-based attachments - this API could be either Android and iOS specific, or it could exist as a universal extension method, that would work on UWP as well, so it could be truly cross-platform.

File: src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs (hunk 2 above)
Comment (MartinZikmund) [reply src/Uno.UWP/ApplicationModel/Email/EmailManager.Android.cs]: {done 👍}

File: doc/ReleaseNotes/_ReleaseNotes.md [hunk 3]
```diff
@@ -1,11 +1,11 @@
 ﻿# Release notes
 
 ### Features
+- Support for `EmailManager.ShowComposeNewEmailAsync`
 - Add support for Automation SetDependencyPropertyValue in Uno.UITest
 - Added support for using a `string` value in a `StaticResource` when using `CreateFromStringAttribute'
 - [Android] Adds support for `FeatureConfiguration.ScrollViewer.AndroidScrollbarFadeDelay`
-- Add support for `Grid.ColumnSpacing` and `Grid.RowSpacing`
-- Add clarification in [documentation](../articles/uno-development/working-with-the-samples-apps.md) for adding automated UI tests
+- [#2040] Support for ms-settings:// special URIs on Android and iOS, Launcher API alignments to match UWP behavior
```
Comment (jeromelaban) [inline doc/ReleaseNotes/_ReleaseNotes.md]: Don't remove release notes ;)

Comment (MartinZikmund) [reply doc/ReleaseNotes/_ReleaseNotes.md]: Oh! That was definitely not intentional 😅

**Comment by MartinZikmund** [comment]
@jeromelaban I see the in progress build already failed due to some Dalvik issue, could you stop and requeue it please :-).? Thank you!

**Comment by zipswich** [comment]
I am a bit confused here.  Prior to embracing Uno, I had been using EmailMessage along with EmailManager to send emails with attachments for a long time:
`wait EmailManager.ShowComposeNewEmailAsync(emailMessage);`

EmailMessage is not implemented.  How should I send attachments with EmailManager?

**Comment by MartinZikmund** [comment]
@zipswitch As `EmailManager` was just merged, it is only available in the [latest preview NuGet package](https://www.nuget.org/packages/Uno.UI/2.1.0-dev.614), so you need to update to it to get access to the API. Regarding attachments - these are currently not yet there, as UWP uses stream-based API whereas Android and iOS work with File-based API - I have issue #2432 opened to discuss potential solutions for this and will implement it next :-) .

**Comment by zipswich** [comment]
@MartinZikmund Thank you for the great work.
My mistake.  I thought I updated Uno.UI immediately after seeing your tweet, but I have just realized that I somehow did not. 
I can see EmailMessage.Attachments is not supported. However, it is still much better than my current way of email for Uno projects: `await Launcher.LaunchUriAsync(uriMail)`.

**Pull request merged**